bin/BuildPackages.sh

```diff
@@ -52,86 +36,203 @@ build_packages() {
 # Even if it doesn't work completely automatically for you, you may get
 # an idea what to do for a complete installation of GAP.
 
+set -e
 
 # Is someone trying to run us from inside the 'bin' directory?
+run_from_bin(){
 if [[ -f gapicon.bmp ]]
 then
   error "This script must be run from inside the pkg directory" \
         "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
+}
+
+# set default parameters
+set_default_parameters() {
 
 # CURDIR
 CURDIR="$(pwd)"
 
-# GAPDIR
-GAPDIR=""
-if [[ "$#" -ge 1 ]]
+# default GAPDIR
+pushd .. >/dev/null
+GAPDIR="$(pwd)"
+popd >/dev/null
+
+# default COLOR
+if [[ -t 1 ]]
 then
+  COLOR=1
+else
+  COLOR=0
+fi
```

Comment:
Why not use "yes" and "no" instead of opaque numerical values? In shell, everything is a string anyway :-).